### PR TITLE
Upgrade to elasticsearch 6.5.1

### PR DESCRIPTION
**Pull request by mfussenegger**
- [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: app/src/main/java/org/elasticsearch/bootstrap/BootstrapProxy.java [hunk 1]
```diff
@@ -182,24 +182,14 @@ private void setup(boolean addShutdownHook, Environment environment) throws Boot
...
 
-        /*
-         * DISABLED setup of security manager due to policy problems with plugins and dependencies.
-         */
-        // install SM after natives, shutdown hooks, etc.
-        //try {
-        //    Security.configure(environment, BootstrapSettings.SECURITY_FILTER_BAD_DEFAULTS_SETTING.get(settings));
-        //} catch (IOException | NoSuchAlgorithmException e) {
-        //    throw new BootstrapException(e);
-        //}
```
Comment (seut) [inline app/src/main/java/org/elasticsearch/bootstrap/BootstrapProxy.java]: I'd prefer leaving it commented out, so everyone knows why this differs from ES Bootstrap and why it is disabled

File: blackbox/docs/general/dml.rst [hunk 2]
```diff
@@ -491,12 +491,12 @@ about failures and successfully imported records is returned.
...
-    | {...} | .../locations1.json |             6 |           0 | {}                                                               |
-    | {...} | .../locations2.json |             5 |           2 | {"failed to parse [date]": {"count": 2, "line_numbers": [1, 2]}} |
-    +--...--+----------...--------+---------------+-------------+------------------------------------------------------------------+
+    +--...--+----------...--------+---------------+-------------+-------------------...--------------------------------------+
+    | node  | uri                 | success_count | error_count | errors                                                     |
+    +--...--+----------...--------+---------------+-------------+-------------------...--------------------------------------+
+    | {...} | .../locations1.json |             6 |           0 | {}                                                         |
+    | {...} | .../locations2.json |             5 |           2 | {"failed to parse ...{"count": 2, "line_numbers": [1, 2]}} |
+    +--...--+----------...--------+---------------+-------------+-------------------...--------------------------------------+
```
Comment (seut) [inline blackbox/docs/general/dml.rst]: isn't this unrelated to the ES upgrade?

File: enterprise/jmx-monitoring/src/main/java/io/crate/plugin/CrateMonitor.java [hunk 3]
```diff
@@ -62,7 +63,7 @@ public CrateMonitor(JobsLogs jobsLogs,
                         ClusterService clusterService,
                         ThreadPool threadPool,
                         CrateCircuitBreakerService breakerService) {
-        logger = Loggers.getLogger(CrateMonitor.class, settings);
+        logger = Loggers.getLogger(CrateMonitor.class, Node.NODE_NAME_SETTING.get(settings));
```
Comment (seut) [inline enterprise/jmx-monitoring/src/main/java/io/crate/plugin/CrateMonitor.java]: isn't the node name now logged by default due to log4j pattern change and logger-with-nodename initialization?

File: blackbox/docs/general/dml.rst (hunk 2 above)
Comment (mfussenegger) [reply blackbox/docs/general/dml.rst]: No, the error message has become more detailed and includes additional information.

File: app/src/main/java/org/elasticsearch/bootstrap/BootstrapProxy.java (hunk 1 above)
Comment (mfussenegger) [reply app/src/main/java/org/elasticsearch/bootstrap/BootstrapProxy.java]: The class already has other differences - I'm not sure if maintaining a commented-out mirror is less confusing as removing the things that don't apply to us. Having a mix of it is definitely not helpful and removing this was less work.

**Comment by mfussenegger** [comment]
@seut pushed a fixup for the logging

**Pull request merged**